### PR TITLE
WinHttpHandler: HttpClientHandlerTestBase.AllowAllCertificates should not be static

### PR DESCRIPTION
**Pull request by antonfirsov**
The default value of this property is `true`. Tests like like `PlatformHandler_HttpClientHandler_ServerCertificates_Http2_Test` are altering this global state, which may cause subsequently executed tests to fail because their prerequisites are not met:
https://github.com/dotnet/runtime/blob/1d9e50cb4735df46d3de0cee5791e97295eaf588/src/libraries/System.Net.Http.WinHttpHandler/tests/FunctionalTests/PlatformHandlerTest.cs#L275-L282

This is the root cause of the failures I discovered in https://github.com/dotnet/runtime/pull/48704#issuecomment-786025511, and a prerequisite to proceed with that PR, since the new tests are incorrect there (although they are all skipped on today's CI machines, so we can't see the failures in the PR).

/cc @geoffkizer @wfurt 

#### Update
Instead of changing the static property to an instance one, `allowAllHttp2Certificates` is now a parameter of the general, shared overload of `CreateHttpClientHandler(...)`. This way individual test classes can implement their mechanism to manage this parameter, so far the only one is `HttpClientHandler_ServerCertificates_Test`.

### CODE REVIEW

File: src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTestBase.cs [hunk 1]
```diff
@@ -40,7 +40,7 @@ public HttpClientHandlerTestBase(ITestOutputHelper output)
 #endif
             };
 
-        protected static HttpClient CreateHttpClient(string useVersionString) =>
+        protected HttpClient CreateHttpClient(string useVersionString) =>
```
Comment (ManickaP) [inline src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTestBase.cs:43]: Some of these methods are static due to being run by `RemoteExecutor`. I don't remember exactly where, but I distinctly remember being bitten by this.
Are you sure this change doesn't break any existing inner and outer loop tests?

Comment (antonfirsov) [reply src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTestBase.cs:43]: Naively, I did only run this with innerloop WinHttpHandler tests. I see now that it breaks the SocketsHttpHandler tests, I will try an another approach I think.

Comment (wfurt) [reply src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTestBase.cs:43]: AFAK methods used with RemoteExecutor do not have to be static. But what ever state the code depends on needs to be marshaled expicitly as there is not magic how to serialize the object to different process.

**Comment by antonfirsov** [comment]
@dotnet/ncl this is ready for review again. Instead of changing the static property to an instance one, `allowAllHttp2Certificates` is now a parameter of the general, shared overload of `CreateHttpClientHandler(...)`. This way individual test classes can implement their mechanism to manage this parameter, so far the only one is `HttpClientHandler_ServerCertificates_Test`.

**Comment by antonfirsov** [comment]
Inner and outerloop test failures are unrelated: #45868, #40798, #46439, #41606 and similar.

**Review by wfurt — APPROVED** [review]
LGTM. 

While this looks ok and solves particular problem I'm wondering about the tests. 
AllowAllCertificates generally skips all TLS cert validation so I'm curious why this is set only for platform handler and why only for HTTP2. I would expect that certificate processing is _not_  dependent on HTTP protocol version.

**Comment by antonfirsov** [comment]
> AllowAllCertificates generally skips all TLS cert validation so I'm curious why this is set only for platform handler and why only for HTTP2. 

I was wondering about the same. @scalablecory @geoffkizer any thoughts on this?

**Pull request merged**